mitodl/micromasters
# Added past course run UI to dashboard

#### What are the relevant tickets?

Closes #939 
Fixes #1121 
Fixes #1122 

#### What's this PR do?

Handles courses with multiple runs per state. This is relevant in cases where the user has failed or missed an upgrade deadline for a course and we want to show them a future run, and also when the user has taken the course before multiple times.

#### Where should the reviewer start?

Probably CourseRow.js as well as dashboard/api.py where a few important changes were made

#### How should this be manually tested?

Here's a [gist](https://gist.github.com/gsidebo/478920b3ead11b4de742c076c2e8d64c) with some functions to help set up test data. Let me know if anything is confusing or doesn't work. Check #939 for the designs that these states should match

#### Screenshots (if appropriate)

![ss 2016-10-13 at 21 23 02](https://cloud.githubusercontent.com/assets/14932219/19372573/42b8a974-918b-11e6-9224-2b0e0eee57fd.png)
